app/models/middleware_server_group.rb

```diff
@@ -0,0 +1,6 @@
+class MiddlewareServerGroup < ApplicationRecord
+  belongs_to :middleware_domain, :foreign_key => "domain_id"
+  belongs_to :lives_on, :polymorphic => true
```

Comment:
Good catch! This shouldn't be there (same holds for the domain entity). It makes sense only for server entity (btw. it's jpkroehling's work that associate the server with a machine)